### ISSUE TITLE
Image normalization

### ISSUE THREAD

**Issue by ririya** [open-mmlab/mmdetection#3932]
Hi everyone I have 2 questions about image normalization. 

1) I always thought that image normalization was about subtracting the mean and dividing by the standard deviation of our own dataset. But from what I see in the different configs, the image normalization config changes when using weights pretrained from different models (example caffe vs pytorch, the caffe based models use std of [1.0,1.0,1.0], the pytorch based models use std of [58.395, 57.12, 57.375]
So my question is: Should image normalization really be based on the pretrained model? 

2) In the image normalization section on the config files, I don't see a normalization by 255. Seems like the images are being passed in the range of 0-255. Is this normalization being computed anywhere that i did not see?

**Comment by MakeCent** [comment]
I am also confused about question2. I didn't find any normalization like /255 neither.
When I print the results before and after Totensor pipeline (line 50 in mmaction.datasets.pipelines.formating):
`        for key in self.keys:
            print('before_totensor: ', results[key].min(), results[key].max())
            results[key] = to_tensor(results[key])
            print('after_totensor: ', results[key].min(), results[key].max())`
I got result like below:
before_totensor:  -2.117904 2.64
before_totensor:  -2.117904 2.64
before_totensor:  -2.117904 2.2489083
before_totensor:  -2.117904 2.4110644
after_totensor:  tensor(-2.1179) tensor(2.6400)
before_totensor:  0.0 1.0
after_totensor:  tensor(-2.1179) tensor(2.6400)
before_totensor:  0.0 1.0
after_totensor:  tensor(0.) tensor(1.)
after_totensor:  tensor(-2.1179) tensor(2.2489)
after_totensor:  tensor(-2.1179) tensor(2.4111)
before_totensor:  0.0 1.0
before_totensor:  0.0 1.0
after_totensor:  tensor(0.) tensor(1.)
after_totensor:  tensor(0.) tensor(1.)
after_totensor:  tensor(0.) tensor(1.)
...

Some of these lines tell me that before or after to_tensor, the min and max are 0 and 1 respectively while the remaining lines show that values in tensor are not ranging from 0 to 1. I don't understand the printed results.

**Comment by ririya** [comment]
@MakeCent 

I think the torchvision models use the normalization you described above. Some models I've used before do first a division by 255 then use mean = [0.485, 0.456, 0.406], std = [0.229, 0.224, 0.225] which results in the same you are using.

**Comment by MakeCent** [comment]
> @MakeCent
> 
> I think the torchvision models use the normalization you described above. Some models I've used before do first a division by 255 then use mean = [0.485, 0.456, 0.406], std = [0.229, 0.224, 0.225] which results in the same you are using.

How stupid I am！So that's just why I didn't find /255 because the mean and std used here have been scaled with *255. That makes the Normalize pipeline do the same things with the operation that first rescale to (0, 1) and then normalize with mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]).

**Comment by ririya** [comment]
Haha I also just realized that. Guess that answers question 2.

About question 1, I noticed most models freeze the backbone, I'm guessing that's the reason why the normalization should stick to the one used on the pretrained model.

**Comment by MakeCent** [comment]
> Haha I also just realized that. Guess that answers question 2.
> 
> About question 1, I noticed most models freeze the backbone, I'm guessing that's the reason why the normalization should stick to the one used on the pretrained model.

Oh, really? Actually, I am quite interested in this because I used to thought the default setting is NOT-freeze weights. For example, backbone [resnet](https://github.com/open-mmlab/mmdetection/blob/d6b1386711b41c21c009f630ab7b5c011a166af6/mmdet/models/backbones/resnet.py#L376) doesn't freeze weights by default. Could you please tell me where you found that most models freeze the backbone?

**Comment by xvjiarui** [comment]
Hi @ririya @MakeCent 
The first layer of ResNet backbone is freeze. And BN is also fixed in the whole backbone. [These lines](https://github.com/open-mmlab/mmdetection/blob/95c05decbe3f9b40de7d312840d7bff261a5f890/configs/_base_/models/mask_rcnn_r50_fpn.py#L10-L12) should explain.

For the question 1, the mean/std is used during backbone pretraining. So we need to use the same mean/std for detection fine-tuning.

**Comment by MakeCent** [comment]
> Hi @ririya @MakeCent
> The first layer of ResNet backbone is freeze. And BN is also fixed in the whole backbone. [These lines](https://github.com/open-mmlab/mmdetection/blob/95c05decbe3f9b40de7d312840d7bff261a5f890/configs/_base_/models/mask_rcnn_r50_fpn.py#L10-L12) should explain.
> 
> For the question 1, the mean/std is used during backbone pretraining. So we need to use the same mean/std for detection fine-tuning.

Thanks for your kind explanation.

**Comment by ririya** [comment]
@xvjiarui on line 11, it says BN is trainable. Am I missing something?


` norm_cfg=dict(type='BN', requires_grad=True),`

**Comment by xvjiarui** [comment]
Hi @ririya 
`norm_eval=True`. 
You may check the docstring [here](https://github.com/open-mmlab/mmdetection/blob/ac20ffef38d6fb244b0e3fd85f20213103732a23/mmdet/models/backbones/resnet.py#L326).

**Comment by ririya** [comment]
@xvjiarui 

Thanks! I was missing that

**Comment by sipie800** [comment]
this is not a detect library, this is a coco library. Normailize everything to coco images, makes no sense at all
